pkg/router/controller/factory/factory.go

```diff
@@ -425,3 +436,29 @@ func (f *RouterControllerFactory) createEndpointSliceSharedInformer() {
 	})
 	f.informers[objType] = informer
 }
+
+func (f *RouterControllerFactory) endpointSliceObjMeta(serviceName string, eps *discoveryv1beta1.EndpointSlice) (*metav1.ObjectMeta, error) {
+	objMeta := eps.ObjectMeta.DeepCopy()
+	objMeta.Name = serviceName
+
+	epType := reflect.TypeOf(&kapi.Endpoints{})
+	item, endpointExists, err := f.informers[epType].GetStore().GetByKey(path.Join(eps.Namespace, serviceName))
+	if err != nil {
+		return nil, err
+	}
+
+	if endpointExists {
+		ep, ok := item.(*kapi.Endpoints)
+		if !ok {
+			return nil, fmt.Errorf("expecting type %q, got %T", epType, ep)
+		}
+		if val, ok := ep.Annotations[unidling.IdledAtAnnotation]; ok && len(val) != 0 {
+			if objMeta.Annotations == nil {
```

Comment:
Worried about dropping any annotations that exist on the endpointslice here. I think the endpointslice mirroring controller applies it's own (and there might be a chance that this code could throw off that controller). Can we just append to the endpointslices current annotations, or return them untouched if we aren't idling?

Reply:
Ah the annotations are just consumed internally so nevermind, no harm in dropping what we dont care about.